### PR TITLE
Feature/ted 985

### PR DESCRIPTION
**Pull request by duprijil**

### CODE REVIEW

File: ted_sws/data_manager/entrypoints/cli/cmd_export_notices_from_mongodb.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        notice_ids=list(notice_id or []),
+        output_folder=output_folder,
+        mongodb_client=mongodb_client
+    )
+    cmd.run()
+
+
+@click.command()
+@click.option('-n', '--notice-id', required=True, multiple=True)
+@click.option('-o', '--output-folder', required=False, default=".")
```
Comment (Dragos0000) [inline ted_sws/data_manager/entrypoints/cli/cmd_export_notices_from_mongodb.py:59]: could we make the CLI to take as optional MONGO_DB_AUTH_URL?

File: ted_sws/data_manager/entrypoints/cli/cmd_export_notices_from_mongodb.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                    self.log(f"Notice {notice_id} saved in {saved_path}.")
+                else:
+                    self.log(f"Notice {notice_id} is not saved.")
+        else:
+            self.log("List with notices is empty.")
+
+        return self.run_cmd_result()
+
+
+def run(notice_id=None,
```
Comment (Dragos0000) [inline ted_sws/data_manager/entrypoints/cli/cmd_export_notices_from_mongodb.py:43]: add documentation to functions

File: ted_sws/data_manager/services/export_notice_from_mongodb.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+
+def save_notice_as_zip(notice: Notice, unpack_path: pathlib.Path):
+    """
+    Serialize notice and saves its information as a .zip file.
+    :param notice:
+    :param unpack_path:
+    """
+
+    def write_in_file(data: Union[str, bytes], terminal_path: str):
```
Comment (Dragos0000) [inline ted_sws/data_manager/services/export_notice_from_mongodb.py:21]: write documentation to functions

**Review by Dragos0000 — APPROVED** [review]
review comments

File: ted_sws/data_manager/services/export_notice_from_mongodb.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        write_path = unpack_path / terminal_path
+        if type(data) == str:
+            write_path.write_text(data=data, encoding="utf-8")
+        elif type(data) == bytes:
+            write_path.write_bytes(data)
+
+    write_in_file(notice.rdf_manifestation.object_data, "rdf_manifestation.ttl")
```
Comment (CaptainOfHacks) [inline ted_sws/data_manager/services/export_notice_from_mongodb.py:28]: In general case rdf_manifestation can be None and here code will fail.

File: ted_sws/data_manager/services/export_notice_from_mongodb.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    write_in_file(notice.distilled_rdf_manifestation.object_data, "distilled_rdf_manifestation.ttl")
```
Comment (CaptainOfHacks) [inline ted_sws/data_manager/services/export_notice_from_mongodb.py:29]: In general case distilled_rdf_manifestation can be None and here code will fail.

File: ted_sws/data_manager/services/export_notice_from_mongodb.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    write_in_file(notice.xml_manifestation.object_data, "xml_manifestation.xml")
```
Comment (CaptainOfHacks) [inline ted_sws/data_manager/services/export_notice_from_mongodb.py:30]: In general case xml_manifestation can be None and here code will fail.

File: ted_sws/data_manager/services/export_notice_from_mongodb.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    write_in_file(notice.validation_summary.object_data, "validation_summary.html")
```
Comment (CaptainOfHacks) [inline ted_sws/data_manager/services/export_notice_from_mongodb.py:31]: In general case validation_summary can be None and here code will fail.

File: ted_sws/data_manager/services/export_notice_from_mongodb.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    write_in_file(json.dumps(notice.xml_metadata.dict()), "xml_metadata.json")
```
Comment (CaptainOfHacks) [inline ted_sws/data_manager/services/export_notice_from_mongodb.py:32]: In general case xml_metadata can be None and here code will fail.

File: ted_sws/data_manager/services/export_notice_from_mongodb.py [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    mets_package_file_name = "mets_manifestation.zip"
+    unpack_mets_package_dir_name = "mets_manifestation"
+    write_in_file(base64.b64decode(notice.mets_manifestation.object_data.encode(encoding="utf-8")),
```
Comment (CaptainOfHacks) [inline ted_sws/data_manager/services/export_notice_from_mongodb.py:35]: In general case mets_manifestation can be None and here code will fail.

File: ted_sws/data_manager/services/export_notice_from_mongodb.py [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+                  mets_package_file_name)
+    mets_package_path = unpack_path / "mets_manifestation.zip"
+    with zipfile.ZipFile(mets_package_path.absolute(), 'r') as zip_ref:
+        zip_ref.extractall(unpack_path / unpack_mets_package_dir_name)
+    for shacl_validation in notice.rdf_manifestation.shacl_validations:
```
Comment (CaptainOfHacks) [inline ted_sws/data_manager/services/export_notice_from_mongodb.py:40]: In general case rdf_manifestation can be None and here code will fail.

File: ted_sws/data_manager/services/export_notice_from_mongodb.py [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+        write_in_file(shacl_validation.object_data, "shacl_validation.html")
+        shacl_validation_json = json.dumps(shacl_validation.validation_results.dict())
+        write_in_file(shacl_validation_json, "shacl_validation.json")
+
+    for sparql_validation in notice.rdf_manifestation.sparql_validations:
```
Comment (CaptainOfHacks) [inline ted_sws/data_manager/services/export_notice_from_mongodb.py:45]: In general case rdf_manifestation can be None and here code will fail.

**Review by CaptainOfHacks — CHANGES_REQUESTED** [review]
Revise comments, fix problems with NoneType fields access subfield ( example notice.rdf_manifestation.shacl_validation) and check any field for None type.

File: ted_sws/data_manager/entrypoints/cli/cmd_export_notices_from_mongodb.py (hunk 1 above)
Comment (kaleanych) [reply ted_sws/data_manager/entrypoints/cli/cmd_export_notices_from_mongodb.py:62]: export_notices_from_mongodb --output-folder=./tmp --notice-id=000163-2021 --mongodb-auth-url="mongodb://user:pass@localhost:27017/"

if mongo-url not provided the .env value will be used

**Review by CaptainOfHacks — APPROVED** [review]

**Pull request merged**